contracts/msig-launcher/src/contract.rs

```diff
@@ -34,216 +32,119 @@ pub fn instantiate(
-                    })?,
-                    admin: Some(CoreModule {}),
-                    funds: vec![],
-                    label: format!("{}-proposal-module", label),
-                }],
-                initial_items: None,
-                dao_uri: None,
-            };
-
-            if PENDING_MSIG.exists(deps.storage) {
-                return Err(ContractError::UnexpectedDoubleTx {});
-            }
-
-            PENDING_MSIG.save(deps.storage, &(label.clone(), info.sender))?;
-
-            Ok(Response::default().add_submessage(SubMsg::reply_on_success(
-                WasmMsg::Instantiate {
-                    admin: None,
-                    code_id: code_ids.main,
-                    msg: to_json_binary(&msg)?,
-                    funds: vec![],
-                    label,
-                },
-                0,
-            )))
-        }
+        } => execute_instantiate(
+            deps,
+            env,
+            info,
+            name,
+            description,
+            image_url,
+            max_voting_period,
+            members,
+        ),
     }
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
     match msg {
-        QueryMsg::MSig { label } => to_json_binary(&MSIG.load(deps.storage, label)?),
-        QueryMsg::CodeIds {} => to_json_binary(&MSIG_CODE_IDS.load(deps.storage)?),
-    }
-}
-
-#[cfg_attr(not(feature = "library"), entry_point)]
-pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
-    let mut resp = Response::default();
-    let code_ids = MSIG_CODE_IDS.load(deps.storage)?;
-    let (label, sender) = PENDING_MSIG.load(deps.storage)?;
-    PENDING_MSIG.remove(deps.storage);
-
-    let mut builder = MSigBuilder::new(sender);
-
-    match msg.result {
-        SubMsgResult::Ok(result) => {
-            for event in result.events {
-                // Look for instantiate type
-                if event.ty == "instantiate" {
-                    // Look for both addr and code id
-                    let mut address = None;
-                    let mut code_id = None;
-                    for attr in event.attributes {
-                        if attr.key == "_contract_address" {
-                            address = Some(attr.value);
-                        } else if attr.key == "code_id" {
-                            code_id = Some(attr.value);
-                        }
-                    }
-
-                    // If both are found, match them into their appropriate address
-                    if let Some((address, code_id)) = address.zip(code_id) {
-                        builder.set_contract(
-                            &code_ids,
-                            code_id.parse::<u64>().unwrap(),
-                            address,
-                        )?;
-                    }
+        QueryMsg::MSigs { pagination } => {
```

Comment:
I'm not sure what frontend expects, but I'd assume filtering by `member` would be useful as well (so someone connects and sees msigs they're featured in). Mentioned it in #3

Reply:
But that's for a future PR

Reply:
Filtering by member would be impossible unless you never change the multisig members

Reply:
Alright, fair. So we'd need to either:
- adjust on the frontend and don't have that list view of a member's featured msigs, but rather just accept a specific address
- or come up with some mechanics when members are updated through a `launcher`'s method (unlikely)